docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,151 @@
+
+For Zeroconf discovery to occur, the Agent's Pod must leverage several Zeroconf depdendencies and libraries. These depdendencies not only expand the size of the Akri Agent (~800MB) but they increase the Agent's surface area and increase the possibility of vulnerabilities.
+
+Discovery is a key functionality of Zeroconf and is straightforward to implement. See the [Browsing services](https://crates.io/crates/zeroconf#browsing-services) examples of the `zeroconf` crate.
+
+One wrinkle is that Akri expects discovery to run to completion. Akri periodically reruns discovery for a protocol. The `zeroconf` crate polls networks for hosts and services.
+
+The implementation used by the protocol is to poll for 5 seconds and report back whichever hosts and services were discovered during that window.
+
+## Broker interfacing
+
+Upon detection, the Akri Zeroconf discovery handler creates an Akri instance to represent each discovered service. The instance contains information about the service using environment variables in each broker pod, if a broker pod is specified in the Configuration.
+
+A more complete description of the `freddie` service could be:
+
+```YAML
+{
+    name: "freddie",
+    kind: "_example._tcp",
+    domain: "local",
+    host_name: "freddie.local",
+    address: "192.168.1.100", port: 8888,
+    txt: ...
+}
+```
+
+Each Broker Pod is configured with environment variables corresponding to the above value:
+
+```bash
+AKRI_ZEROCONF_DEVICE_KIND=_example._tcp
+AKRI_ZEROCONF=zeroconf
+AKRI_ZEROCONF_DEVICE_HOST=freddie.local
+AKRI_ZEROCONF_DEVICE_NAME=freddie
+AKRI_ZEROCONF_DEVICE_PORT=8888
+AKRI_ZEROCONF_DEVICE_ADDR=192.168.1.100
+```
+
+The service's TXT records are not provided by the implementation of the discovery handler. The TXT records could be enumerated as additional environment variables but it's unclear how best to represent these, possibly: `AKRI_ZEROCONF_DEVICE_[[KEY]]=[[VALUE]]`.
+
+The Broker sample enumerates these values to standard output every 5 seconds.
+
+In practice, a Kubernetes application would use this data to identify services and invoke them.
+
+## Security Considerations
+
+The Akri Agent is only able to browse services accessible to its cluster's hosts.
+
+There are no security considerations for service (mDNS) browsing. This functionality is akin to DNS lookups.
+
+Accessing services that are found by service browsing *may* require the provision of credentials. This would be an implementation detail of an Akri Broker and is not considered further here.
+
+## Outstanding Questions
+
++ What would a generic Akri Zeroconf Broker do? In practice, the application developer would likely wish to implement the Broker for their specific application.
```

Comment:
this is not unlike udev.  we don't provide a generic udev broker, but have a udev video broker.  perhaps a similar approach can be applied here?